components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/model/UserStoreAttributeMappings.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    /**
+     * Get attribute mappings changes for the given user store type.
+     *
+     * @param userStoretype User store type.
+     * @return Map of attribute uri and attribute mappings of the given user store type.
+     */
+    public Map<String, UserStoreAttributeDO> getUserStoreAttributeMappings(String userStoretype) {
+
+        if (userStoreAttrMappings.containsKey(userStoretype)) {
+            return Collections.unmodifiableMap(userStoreAttrMappings.get(userStoretype));
+        }
+        /*
+         * If details are not available for the given user store type, return default values.
+         * Add a new userstore attribute mappings file to conf/attributes/userstore directory to
+         * configure correct attribute mappings for the userstore.
+         */
+        LOG.warn(String.format("No record found for the given userstore: %s. Default attribute mappings are " +
+                        "returning. Please add userstore attribute mappings file for the userstore %s " +
+                        "to conf/attributes/userstore directory to configure correct attribute mappings.",
+                userStoretype, userStoretype));
+        return defaultUserStoreAttrMapping;
+    }
+
+    /**
+     * Set all available user store attribute mappings.
+     *
+     * @param availableUserStoreAttrMappings Map of user store type and their attribute mappings changes.
+     */
+    public void setUserStoreAttributeMappings(Map<String, Map<String, ChangedUserStoreAttributeDO>>
+                                                      availableUserStoreAttrMappings) {
+
+        for (String userStoreType : availableUserStoreAttrMappings.keySet()) {
+            Map<String, UserStoreAttributeDO> tempMap = getModifiedAttributeMap(availableUserStoreAttrMappings
+                    .get(userStoreType));
+            this.userStoreAttrMappings.put(userStoreType, tempMap);
+        }
+    }
+
+    /**
+     * To merge default attribute mappings and mappings changes of other user stores.
+     *
+     * @param changedUserStoreAttrMap Attribute mapping changes which should change default values.
+     * @return Map of attribute mappings.
+     */
+    private Map<String, UserStoreAttributeDO> getModifiedAttributeMap(Map<String, ChangedUserStoreAttributeDO>
+                                                                              changedUserStoreAttrMap) {
+
+        if (defaultUserStoreAttrMapping == null) {
+            return null;
+        }
+        Gson gson = new Gson();
+        String serializedDefaultAttrMappings = gson.toJson(defaultUserStoreAttrMapping);
+        // To deserialize a hashmap using Gson, need a type object of the hashmap.
+        Type type = new TypeToken<HashMap<String, UserStoreAttributeDO>>() {
+        }.getType();
+        Map<String, UserStoreAttributeDO> clonedAttrMap = gson.fromJson(serializedDefaultAttrMappings, type);
+        for (String claimId : changedUserStoreAttrMap.keySet()) {
```

Comment:
This method accesses the value of a Map entry, using a key that was retrieved from a keySet iterator. It is more efficient to use an iterator on the entrySet of the map, to avoid the Map.get(key) lookup.

Reply:
fixed in https://github.com/wso2/carbon-identity-framework/pull/3715